### ISSUE TITLE
"Include untracked files in stash" not working?

### ISSUE THREAD

**Issue by michele-amati** [gitextensions/gitextensions#7464]
## Current behaviour
Regardless of this option:
![image](https://user-images.githubusercontent.com/12916292/69792666-5296cc80-11c7-11ea-9262-453b55370f80.png)
When I stash something with this menu:
![image](https://user-images.githubusercontent.com/12916292/69792784-8b36a600-11c7-11ea-82f2-28e4f6af5f3b.png)
Untracked files are not stashed. They do not appear in the stash, and do not disappear from the "Working directory changes" area.

## Expected behaviour
With that option selected, I expected stash to work on untracked files too, like the command `git stash -u` (which works)

## Steps to reproduce

Have one or more untracked files and try to stash them with and without that option selected, you'll see that they stays there, unstashed, in both cases.

## Did this work in previous version of GitExtensions

I've never seen this works, but first time I tried has been in 2018

## Environment

- Git Extensions 3.2.1.6628
- Build bc07663c3c893f1d896520e0bdcfbe35cfe4b729
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4018.0
- DPI 96dpi (no scaling)

## Similar issues

I've found few issues that may be linked to the problem, but they are very old and mentions options that I can't find in the current version. 

- https://github.com/gitextensions/gitextensions/issues/875
- https://github.com/gitextensions/gitextensions/issues/1300
- https://github.com/gitextensions/gitextensions/pull/794

**Comment by gerhardol** [comment]
Works for me. Check the GE Git Command Log (F12)

**Comment by michele-amati** [comment]
With and without that option, the command executed is always: `git stash save`

**Comment by michele-amati** [comment]
Kind of solved it, this is from the `GitExtensions.settings` file:
```xml
<item>
  <key>
    <string>includeUntrackedFilesInAutoStash</string>
  </key>
  <value>
    <string>true</string>
  </value>
</item>
<item>
  <key>
    <string>includeUntrackedFilesInManualStash</string>
  </key>
  <value>
    <string>false</string>
  </value>
</item>
<item>
```
- `includeUntrackedFilesInAutoStash` is the variable behind the "Include untracked files in stash" gui option
- `includeUntrackedFilesInManualStash` I wasn't able to find the gui equivalent of this one, but this is the one that does what I wanted, for now I just changed it in the config file, can anyone point me to where this option is in the gui?

**Comment by pmiossec** [comment]
>can anyone point me to where this option is in the gui?

Wasn't the one in the "Manages stashes..." menu ?
![image](https://user-images.githubusercontent.com/460196/69980381-6aa47e00-1530-11ea-9d86-6bad8f63806d.png)

![image](https://user-images.githubusercontent.com/460196/69980365-624c4300-1530-11ea-9be8-70131f473065.png)

**Comment by gerhardol** [comment]
The autostash variant is used in other situations.

From this report, it could be informative to have the manual option in Settings (in addition to FormStash). If you never use the stash form, it can be hard to find the option.
The problem is that the Settings-Appearance form is quite busy at is already...

**Comment by RussKie** [comment]
I'd start with a question - why do we need two different switches? And is there a reason we can't use one switch?

> The problem is that the Settings-Appearance form is quite busy at is already...

I don't believe "Appearance" is the right widget page.

**Comment by michele-amati** [comment]
> > can anyone point me to where this option is in the gui?
> 
> Wasn't the one in the "Manages stashes..." menu ?

And now I feel like an idiot... Thanks that's exactly the option I was looking for.

To be totally honest though, the location/naming of those settings is not very intuitive. The separation of stash in `manual` and `auto` is not an official git thing AFAIK, I had no idea it existed. My logical steps has been:

1. see untracked files are not stashed
2. search for "stash" in settings
3. find 1 and just 1 option that is titled exactly like what I wanted to achive "Include untracked files in stash"
4. try angain and it does not work

Maybe the option in settings can be renamed to clearly state that it affects automatic stash only, so that anyone reaching it at least knows he should search elsewhere for the manual version.

**Comment by pmiossec** [comment]
>Maybe the option in settings can be renamed to clearly state that it affects automatic stash only

Definitely.

**Comment by RussKie** [comment]
Why do we need to separate `manual` vs `auto`?

**Comment by gerhardol** [comment]
> Maybe the option in settings can be renamed to clearly state that it affects automatic stash only

The message already is _Include untracked in autostash_ in Settings->Behaviour (where similar options are located already). But there is no manual to compare to.
FormStash has _Include untracked files_ option, could have been  _Include untracked in manual_

It may be better to remove the setting in Settings
(there may be situations where the settings do not show up though)

> Why do we need to separate manual vs auto?

auto: what is automatically stashed at operations like checkout. untracked is preferred here, to avoid conflicts, the information should be restored anyway

manual: what you include when creating a stash. You may not want to include untracked here

**Comment by michele-amati** [comment]
> > Maybe the option in settings can be renamed to clearly state that it affects automatic stash only
> 
> The message already is _Include untracked in autostash_ in Settings->Behaviour (where similar options are located already). But there is no manual to compare to.

indeed it is https://github.com/gitextensions/gitextensions/commit/412a9f5b7433e336e9513857024ee4d7cc036aee#diff-c77f8725ed4e8257ca706ec8f3363704
apparently the change is not yet in the version `Git Extensions 3.2.1.6628` that is the latest you can get with the installer

**Comment by RussKie** [comment]
Closing as it was fixed in 3.3
You can try the RC1 now
